### PR TITLE
docs: add examples for Login, Logout and Credential

### PR DESCRIPTION
**Pull request by wangxiaoxuan273**
Part of #49

### CODE REVIEW

File: example_test.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	cred := auth.Credential{Username: "username-example", Password: "password-example"}
+	err = credentials.Login(context.Background(), store, registry, cred)
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println("Login succeeded")
+}
+
+func ExampleLogout() {
+	store, err := credentials.NewStore("example/path/config.json", credentials.StoreOptions{AllowPlaintextPut: true})
```
Comment (Wwwsylvia) [inline example_test.go:44]: Let's leave the StoreOptions empty.

File: example_test.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	}
+	err = credentials.Logout(context.Background(), store, "example.registry.com")
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println("Logout succeeded")
+}
+
+func ExampleCredential() {
+	store, err := credentials.NewStore("example/path/config.json", credentials.StoreOptions{AllowPlaintextPut: true})
```
Comment (Wwwsylvia) [inline example_test.go:56]: Same here.

File: example_test.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+	client := auth.Client{}
+	client.Credential = credentials.Credential(store)
+
+	request, err := http.NewRequest(http.MethodGet, "localhost:8080", nil)
+	if err != nil {
+		panic(err)
+	}
+
+	client.Do(request)
```
Comment (Wwwsylvia) [inline example_test.go:69]: nit: Do we need to check the error?

File: example_test.go (hunk 1 above)
Comment (wangxiaoxuan273) [reply example_test.go:44]: done

File: example_test.go (hunk 2 above)
Comment (wangxiaoxuan273) [reply example_test.go:56]: done.

File: example_test.go (hunk 3 above)
Comment (wangxiaoxuan273) [reply example_test.go:69]: added error check

File: example_test.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+	fmt.Println("Logout succeeded")
+}
+
+func ExampleCredential() {
+	store, err := credentials.NewStore("example/path/config.json", credentials.StoreOptions{})
+	if err != nil {
+		panic(err)
+	}
+
+	client := auth.Client{}
```
Comment (Wwwsylvia) [inline example_test.go:61]: Might be better to use `auth.DefaultClient`

Comment (wangxiaoxuan273) [reply example_test.go:61]: Forgot about its existence.

**Review by Wwwsylvia — APPROVED** [review]
LGTM

File: example_test.go [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+func ExampleLogin() {
+	store, err := credentials.NewStore("example/path/config.json", credentials.StoreOptions{AllowPlaintextPut: true})
+	if err != nil {
+		panic(err)
+	}
+	registry, err := remote.NewRegistry("example.registry.com")
+	if err != nil {
+		panic(err)
+	}
+	cred := auth.Credential{Username: "username-example", Password: "password-example"}
```
Comment (shizhMSFT) [inline example_test.go:35]: Let's format the code.

File: example_test.go [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+	"oras.land/oras-go/v2/registry/remote"
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+func ExampleLogin() {
+	store, err := credentials.NewStore("example/path/config.json", credentials.StoreOptions{AllowPlaintextPut: true})
+	if err != nil {
+		panic(err)
+	}
+	registry, err := remote.NewRegistry("example.registry.com")
```
Comment (shizhMSFT) [inline example_test.go:31]: I think `ghcr.io` or `localhost:5000` would be a better example registry.

File: example_test.go (hunk 5 above)
Comment (wangxiaoxuan273) [reply example_test.go:35]: Formatted.

File: example_test.go (hunk 6 above)
Comment (wangxiaoxuan273) [reply example_test.go:31]: changed to `localhost:5000`.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**